app/components/SignupForm.tsx

```diff
@@ -1,32 +1,140 @@
+  validateRequired,
+} from 'utils/validate';
+
+interface FormType {
+  id: string;
+  name: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+}
+
+const INITIAL_DATA = {
+  id: '',
+  name: '',
+  email: '',
+  password: '',
+  confirmPassword: '',
+};
 
 const SignupForm: React.FC = () => {
+  const [formData, setFormData] = useState<FormType>(INITIAL_DATA);
+  const [errorText, setErrorText] = useState<FormType>(INITIAL_DATA);
+
   const handleSubmit = async (e: React.FormEvent) => {
     e.preventDefault();
+    if (confirmFormData()) {
+      const userInfos = JSON.parse(localStorage.getItem('users') || '[]');
+      userInfos.push(formData);
+      localStorage.setItem('users', JSON.stringify(userInfos));
+    }
+  };
+
+  const handleChangeInput = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+
+  const confirmFormData = () => {
+    const newErrors = { ...INITIAL_DATA };
+    const { id, name, email, password, confirmPassword } = formData;
+    let result;
+
+    result = validateRequired(id);
+    if (result.isValid) {
+      const minLengthCheck = validateMinLength(id, 5);
+      const maxLengthCheck = validateMaxLength(id, 15);
+
+      if (!minLengthCheck.isValid) {
+        newErrors.id = minLengthCheck.message;
+      } else if (!maxLengthCheck.isValid) {
+        newErrors.id = maxLengthCheck.message;
+      }
+    } else {
+      newErrors.id = result.message;
+    }
+
+    result = validateRequired(name);
+    if (!result.isValid) newErrors.name = result.message;
+
+    result = validateEmail(email);
+    if (!result.isValid) newErrors.email = result.message;
+
+    result = validateRequired(password);
+    if (!result.isValid) {
+      newErrors.password = result.message;
+    } else {
+      result = validatePasswordFormat(password);
+      if (!result.isValid) {
+        newErrors.password = result.message;
+      } else {
+        result = validateMinLength(password, 8);
+        if (!result.isValid) {
+          newErrors.password = result.message;
+        } else {
+          result = validateMaxLength(password, 20);
+          if (!result.isValid) {
+            newErrors.password = result.message;
+          }
+        }
+      }
+    }
+
+    result = validateRequired(confirmPassword);
+    if (!result.isValid) {
+      newErrors.confirmPassword = result.message;
+    } else {
+      result = validatePasswordMatch(password, confirmPassword);
+      if (!result.isValid) {
+        newErrors.confirmPassword = result.message;
+      }
+    }
+
+    setErrorText(newErrors);
+    return Object.keys(newErrors).every((key) => newErrors[key as keyof FormType] === '');
   };
 
   return (
     <form onSubmit={handleSubmit}>
       <div>
         <label htmlFor="id">ID:</label>
-        <input id="id" />
+        <input id="id" value={formData.id} onChange={handleChangeInput} />
```

Comment:
```suggestion
        <input id="id" name="id" value={formData.id} onChange={handleChangeInput} />
```
name 어트리뷰트가 빠져있습니다. 나머지 input들도 name 속성을 추가해주면 easy 테스트는 모두 통과합니다.